be/src/formats/parquet/complex_column_reader.cpp

```diff
@@ -105,6 +105,34 @@ Status ListColumnReader::read_range(const Range<uint64_t>& range, const Filter*
     return Status::OK();
 }
 
+Status ListColumnReader::fill_dst_column(ColumnPtr& dst, ColumnPtr& src) {
+    ArrayColumn* array_column_src = nullptr;
+    ArrayColumn* array_column_dst = nullptr;
+    if (src->is_nullable()) {
+        NullableColumn* nullable_column_src = down_cast<NullableColumn*>(src.get());
+        DCHECK(nullable_column_src->mutable_data_column()->is_array());
+        array_column_src = down_cast<ArrayColumn*>(nullable_column_src->mutable_data_column());
+        NullColumn* null_column_src = nullable_column_src->mutable_null_column();
+        NullableColumn* nullable_column_dst = down_cast<NullableColumn*>(dst.get());
+        DCHECK(nullable_column_dst->mutable_data_column()->is_array());
+        array_column_dst = down_cast<ArrayColumn*>(nullable_column_dst->mutable_data_column());
+        NullColumn* null_column_dst = nullable_column_dst->mutable_null_column();
+        null_column_dst->swap_column(*null_column_src);
+        nullable_column_src->update_has_null();
+        nullable_column_dst->update_has_null();
+    } else {
+        DCHECK(src->is_array());
+        DCHECK(dst->is_array());
+        DCHECK(!get_column_parquet_field()->is_nullable);
+        array_column_src = down_cast<ArrayColumn*>(src.get());
+        array_column_dst = down_cast<ArrayColumn*>(dst.get());
+    }
+    array_column_dst->offsets_column()->swap_column(*(array_column_src->offsets_column()));
```

Comment:
Why not use ArrayColumn::swap_column?

Reply:
we only swap offset and nullcolumn, for element_column, we fill it with element_reader